command.go

```diff
@@ -115,57 +115,30 @@ func (c Command) Run(ctx *Context) (err error) {
 		return err
 	}
 	set.SetOutput(ioutil.Discard)
-
+	firstFlagIndex, terminatorIndex := getIndexes(ctx)
```

Comment:
I liked that you started pulling this logic into functions. What do you think about also pulling this whole flag handling logic into a separate function? It'd allow us to do things like return early to reduce nesting as well as compartmentalize this handling.

Reply:
I would have done more, but I couldn't logically see a path through this.  I was thinking in V2 it might be easier ?

Reply:
👍 we can always do it as a second pass.